src/libsyntax/parse/parser.rs

```diff
@@ -6007,7 +6007,9 @@ impl<'a> Parser<'a> {
         }
 
         if self.check_keyword(keywords::Const) {
-            return Err(self.span_fatal(self.span, "extern items cannot be `const`"));
+            let mut err = self.span_fatal(self.span, "extern items cannot be `const`");
+            err.help("use `static` instead");
+            return Err(err);
```

Comment:
That would be better, but I just wanted to improve the error message. I don't feel the need to rework the error handling here.